src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -240,27 +252,20 @@ private static string GetAllExceptionMessages(Exception exception)
             return exceptionMessage;
         }
 
-        private string GenerateResultString(Types.IEvaluation evaluation)
+        private IReportWriter GetReportWriter(string reportFormat, FileInfo outputFile, string rootFolder = null)
         {
-            string resultString = "";
-
-            if (!evaluation.Passed)
+            if (Enum.TryParse<ReportFormat>(reportFormat, ignoreCase:true, out ReportFormat format))
             {
-                foreach (var result in evaluation.Results)
+                switch (format)
                 {
-                    if (!result.Passed)
-                    {
-                        resultString += $"{TwiceIndentedNewLine}Line: {result.LineNumber}";
-                    }
-                }
-
-                foreach (var innerEvaluation in evaluation.Evaluations)
-                {
-                    resultString += GenerateResultString(innerEvaluation);
+                    case ReportFormat.Sarif:
+                        return new SarifReportWriter((FileInfoBase)outputFile, rootFolder);
+                    case ReportFormat.Console:
```

Comment:
I think it would actually be better to remove `default`, so that when a new value is added, the compiler will force the author to choose an action here.

Reply:
removed default branch